task21/js/main.js

```diff
@@ -0,0 +1,154 @@
+    const tableHeaderFragment = document.createDocumentFragment();
+    for(const column of Object.values(userTableColumn)){
+        const tableHeaderElement = createElementWithClassName("th", "table-header");
+        tableHeaderElement.classList.add("bg-gray-900", "text-white");
+        tableHeaderElement.textContent = column;
+        column === "ID" && tableHeaderElement.appendChild(getCreatedSortButton());
+        tableHeaderFragment.appendChild(tableHeaderElement);
+    }
+    theadElement.appendChild(tableRowElement).appendChild(tableHeaderFragment);
+    return theadElement;
+}
+
+const getCreatedTableBody = (userContentsData) => {
+    const tbodyElement = document.createElement("tbody");
+    tbodyElement.id = "js-tableBody";
+    const tableBodyFragment = document.createDocumentFragment();
+    for(const user of userContentsData){
+        const tableRowElement = document.createElement("tr");
+        for(const column of Object.keys(userTableColumn)){
+            const tableDataElement = createElementWithClassName("td", "table-data");
+            tableDataElement.textContent = user[column];
+            tableBodyFragment.appendChild(tableRowElement).appendChild(tableDataElement);
+        }
+    }
+    tbodyElement.appendChild(tableBodyFragment);
+    return tbodyElement;
+};
+
+const getCreatedSortButton = () => {
+    const sortButton = document.createElement("button");
+    const sortImage = createElementWithClassName("img", "sort-image");
+    sortButton.id = "js-sortButton";
+    sortImage.dataset.status = "default";
+    sortButton.appendChild(sortImage);
+    return sortButton;
+};
+
+const addEventForSortButton = (userContentsData) => {
+    const sortButton = document.getElementById("js-sortButton");
+    sortButton.addEventListener('click', (e) => {
+        const nextStatus = changeSortStatus(e.target.dataset.status);
+        e.target.dataset.status = nextStatus;
+
+        sortContetns(userContentsData, nextStatus);        
+    })
+}
+
+const changeSortStatus = (status) => {
+    switch(status){
+        case "default":
+            return "asc";
+        case "asc":
+            return "desc";
+        case "desc":
+            return "default";
+        default:
+            return "default";
+    }
+}
+
+const renderTableData = (userContentsData) => {
+    document.getElementById("js-tableBody").remove();
+    const table = document.querySelector("table");
+    table.appendChild(getCreatedTableBody(userContentsData));
+};
+
+const sortContetns = (userContentsData, status) => {
+    const cloneUserData = [...userContentsData];
+    if(status === "default") return renderTableData(userContentsData);
+    switch(status){
+        case "asc":
+            cloneUserData.sort((a, b) => a.id - b.id);
+            renderTableData(cloneUserData);
+        break;
+        case "desc":
+            cloneUserData.sort((a, b) => b.id - a.id);
+            renderTableData(cloneUserData);
+    }
+}
```

Comment:
nits
I thought it could be summarized in the SWITCH statement. However, I think it is a matter of preference, so I will leave the modification to you.


```suggestion
const sortContetns = (userContentsData, status) => {
  const cloneUserData = [...userContentsData];
  switch (status) {
    case "asc":
      cloneUserData.sort((a, b) => a.id - b.id);
      renderTableData(cloneUserData);
      break;
    case "desc":
      cloneUserData.sort((a, b) => b.id - a.id);
      renderTableData(cloneUserData);
      break;
    default:
      return renderTableData(userContentsData);
  }
};
```

Reply:
I agree with @csakyo san.
I thought it would be easier to understand if you put it in a Switch sentence!